pgx-utils/src/sql_entity_graph/pg_extern/mod.rs

```diff
@@ -478,8 +471,19 @@ impl PgExtern {
                                 funcctx.user_fctx = pgx::PgMemoryContexts::For(funcctx.multi_call_memory_ctx).palloc_struct::<IteratorHolder<#retval_ty_resolved>>() as *mut ::core::ffi::c_void;
                                 iterator_holder = pgx::PgBox::from_pg(funcctx.user_fctx as *mut IteratorHolder<#retval_ty_resolved>);
 
-                                #( #arg_fetches )*
-                                #result_handler
+                                // function arguments need to be "fetched" while in the function call's
+                                // multi-call-memory-context to ensure that any detoasted datums will
+                                // live long enough for the SRF to use them over each call
```

Comment:
This is what needs to also be documented in a place outside the context of this function's comments. Especially so because this is something that will always need to remain true: we always must satisfy this sort of lifetime-binding condition, and it doesn't matter whether the code is burned and rewritten a million times: it remains a constraint. It should be easier to reference this sort of information than reading thousands of lines of implementations line by line, because it determines how fast someone can begin to understand the concepts required to write correct code for implementing PGX.